### PR TITLE
Refactor Cases

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR refactors Cases.jl by:
 - Removing unnecessary `theta` fields
 - Using local variables to reduce `getproperty` calls
 - Fuses some loops

So far, this branch only applies this pattern to a small set of Cases.jl, but it'd be nice if someone could take this over and apply to the rest of the file.

I know that this may interfere with the thermodynamics PR (one reason why I didn't finish), so maybe we can wait until that merges, first, before continuing with this.

### CODE REVIEW

File: integration_tests/utils/Cases.jl [hunk 1]
```diff
@@ -403,32 +389,36 @@ function initialize_forcing(self::CasesBase{Bomex}, grid::Grid, ref_state::Refer
...
+        if zc_k <= 300.0
+            dqtdt = -1.2e-8   #kg/(kg * s)
+        elseif zc_k > 300.0 && zc_k <= 500.0
+            dqtdt = -1.2e-8 + (zc_k - 300.0) * (0.0 - -1.2e-8) / (500.0 - 300.0) #kg/(kg * s)
+        else
+            dqtdt = 0.0
         end
+        self.Fo.dqtdt[k] = dqtdt
 
         #Set large scale subsidence
```
Comment (yairchn) [inline integration_tests/utils/Cases.jl:414]: subsidence = 0.0

Comment (charleskawczynski) [reply integration_tests/utils/Cases.jl:414]: I'm not sure I understand this comment

Comment (charleskawczynski) [reply integration_tests/utils/Cases.jl:414]: Ah, by default, yes.

Comment (yairchn) [reply integration_tests/utils/Cases.jl:414]: above 'zc_k <= 2100.0' --> subsidence = 0.0

**Comment by charleskawczynski** [comment]
I don't like that these changes are making this file even larger. In the long term, when more loops are fused, this packing/unpacking pattern will probably pay off, but I'm not sure it's paying off with the limited scope at the moment. I may try to peel off pieces of this PR that seem less iffy

**Comment by charleskawczynski** [comment]
Superseded by #302

**Pull request closed without merging**